tests/test_spark.py

```diff
@@ -77,25 +77,54 @@ def test_teragen():
 
 @pytest.mark.sanity
 def test_supervise():
-    spark_drivers = shakedown.get_service_ips("spark")
-    assert len(spark_drivers) == 0, "Shouldn't be any spark drivers running, got {}".format(len(spark_drivers))
+    def streaming_job_registered():
+        return shakedown.get_service("HdfsWordCount") is not None
+
+    def streaming_job_is_not_running():
+        return not streaming_job_registered()
+
+    def has_running_executors():
+        f = shakedown.get_service("HdfsWordCount")
+        if f is None:
+            return False
+        else:
+            return len([x for x in f.dict()["tasks"] if x["state"] == "TASK_RUNNING"]) > 0
+
     driver_id = utils.submit_job(app_url=SPARK_EXAMPLES,
                                  app_args="file:///mnt/mesos/sandbox/",
                                  app_name="/spark",
                                  args=["--supervise",
                                        "--class", "org.apache.spark.examples.streaming.HdfsWordCount",
-                                       "--conf", "spark.cores.max=1",
-                                       "--conf", "spark.executors.cores=1"])
+                                       "--conf", "spark.cores.max=8",
+                                       "--conf", "spark.executors.cores=4"])
     LOGGER.info("Started supervised driver {}".format(driver_id))
-    utils.wait_for_executors_running("HdfsWordCount", 1)
-    spark_drivers = shakedown.get_service_ips("spark")
-    assert len(spark_drivers) == 1, "Should be 1 spark driver running got {}".format(len(spark_drivers))
-    driver_ip = spark_drivers.pop()
-    shakedown.kill_process_on_host(hostname=driver_ip, pattern=driver_id)
-    utils.wait_for_executors_running("HdfsWordCount", 1, 1200)
+    shakedown.wait_for(lambda: streaming_job_registered(),
+                       ignore_exceptions=False,
+                       timeout_seconds=600)
+    LOGGER.info("Job has registered")
+    shakedown.wait_for(lambda: has_running_executors(),
+                       ignore_exceptions=False,
+                       timeout_seconds=600)
+    LOGGER.info("Job has running executors")
+
+    host = shakedown.get_service("HdfsWordCount").dict()["hostname"]
+    id = shakedown.get_service("HdfsWordCount").dict()["id"]
+    driver_regex = "spark.mesos.driver.frameworkId={}".format(id)
+    shakedown.kill_process_on_host(hostname=host, pattern=driver_regex)
```

Comment:
I don't understand how this pattern matching works. The pattern should match the name of the process, but I doubt that the process name has 'spark.mesos.driver.frameworkId' in it?

Reply:
Oh, is it specified by the dispatcher?

Reply:
Yes. I added the additional `spark.mesos.driver.frameworkId` so that we don't kill some of the infrastructure processes, there are some logging processes that have the driver string and without the additional pattern those get killed also.